guides/being-a-tech-lead-at-dxw.md

```diff
@@ -0,0 +1,149 @@
+don't know how much software costs.
+
+That means part of your role is education. For the public sector to properly
+complete their digital transformation, it's not enough to put some services on
+the internet. We need to reverse that loss of expertise, so that all the folks
+involved in building services are able to make well informed decisions. As a
+tech lead:
+
+- you understand how software is built
+- you understand how to work in an agile way
+- hopefully you understand how much software costs (though we're always happy to
+  teach you how to think about this)
+
+Share that understanding.
+
+Another feature of public sector technology (though by no means unique to the
+public sector) is a lack of money to compete with big tech and venture capital
+when hiring. That means there can be a lack of experienced technologists, and a
+short tenure for those technologists they do manage to hire. But it also means
+you'll often be working with folks who are motivated more by doing public good
+than they are by chasing money.
+
+### ...at the intersection of the two
+
+When building a public service, it can be hard to ever call it done. Especially
+when much of your team's time is spent dealing with broken foundations (lack of
+APIs, legacy systems, or messy and incomplete data) rather than building the
+service. But the lack of money in the public sector means we often have to
+support our clients in building something that's good enough, before they
+redirect their resources onto other priorities. You can serve the users and
+client best by building in an agile way throughout. Always find the minimal
+version of the feature your team can ship, and leave it open for iteration
+later. With luck you'll get to do that iteration, but if you don't our client
+will have a service that's always in a workable state no matter where we stop.
+
+Governance is generally a good thing when built from an understanding of how to
+work in an agile way and how software is built. When it comes from a history of
+supplier management, it can instead smell a lot like waterfall approaches and
+arse covering. If you're in this situation then building relationships becomes
+even more important. You'll need to engage with the governance processes, but
+you'll also need to push back against parts of them. It's likely that the client
+is undergoing something of a digital transformation, so there are probably
+people there who can help you push an agile approach. You're not on your own in
+doing this; get other folks in the team (like your delivery lead or technical
+architect) to help, and call on the wider dxw team when you need it.
+
+## What our tech leads do inside teams
+
+The role of a tech lead at dxw is to lead the technical work of the team. Given
+software is built by and for people, that means it's very much at the squishy
+intersection of technical expertise and people skills.
+
+A tech lead's job is to:
```

Comment:
Some further ideas:

- advocate building software that meets user needs
- teach others directly and by example
- be responsible for the team learning from failure, ensure incident reviews happen and that the team has the opportunity to learn

Reply:
@tahb good point about emphasising the "user-driven" viewpoint. I've found it's often helpful to bring teams' focus back to the user benefit we're trying to deliver.

Perhaps there could be a bullet point about 'quality control' which might follow the "teach others directly and by example" item e.g.

- encourage team members to go the extra 'yard' to make their work clear and maintainable

Reply:
Adding some bullets to cover these points.